aioxmpp/avatar/service.py

```diff
@@ -206,7 +211,8 @@ class AbstractAvatarDescriptor:
     """
 
     def __init__(self, remote_jid, mime_type, id_, nbytes, width=None,
```

Comment:
Does it make sense to have all the attributes on the **Abstract**AvatarDescriptor by default? I would’ve expected it to define an interface with abstract properties maybe and let subclasses deal with things.

Especially the ``vcard`` which is exclusively used by the ``VCardAvatarDescriptor`` should be moved there.

``image_bytes`` seems to be unused.

Reply:
Image bytes is used in the case that the cache is empty and we call get_avatar_metadata which gets the vcard. To prevent another roundtrip we set the image bytes property.

The interface where abstract avatar has all the properties comes from the times when we only supported PEP avatars (where this made much more sense). I think I will have to break the interface slightly, to improve this, but since the constructors of the avatar descriptors are not part of the public interface this should not matter.

Reply:
Ah, right, I missed that somehow (and I assumed that ``_vcard`` referred to the actual vcard). Nevermind then.

(Even though PEP 484 type annotations would be great here.)